oauth-common/src/main/java/io/strimzi/kafka/oauth/common/HttpUtil.java

```diff
@@ -45,6 +46,44 @@ public class HttpUtil {
     static final int DEFAULT_CONNECT_TIMEOUT = getConnectTimeout(new Config());
     static final int DEFAULT_READ_TIMEOUT = getReadTimeout(new Config());
 
+    public static <T> T doWithRetries(int retries, long retryPauseMillis, IOTask<T> task) throws ExecutionException {
```

Comment:
All the callers of this method all to essentially the same thing, which is wrap a call to a request method (`get` or `post`) in a lambda _which also maintains some metrics_. I wonder if it would make sense to make the metrics part of this method.

Reply:
I added `MetricsHandler` to `doWithRetries` method. It made the code cleaner in places of use.
Great suggestion, thanks.